pkg/kubectl/cmd/clusterinfo/clusterinfo_dump_test.go

```diff
@@ -68,3 +79,109 @@ func TestSetupOutputWriterFile(t *testing.T) {
+	defer tf.Cleanup()
+
+	ns := scheme.Codecs
+
+	encodeResp := func(obj runtime.Object, ver runtime.GroupVersioner) io.ReadCloser {
+		info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
+		encoder := ns.EncoderForVersion(info.Serializer, ver)
+
+		return ioutil.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(encoder, obj))))
+	}
+
+	encodeRespForCoreGV := func(obj runtime.Object) io.ReadCloser {
+		return encodeResp(obj, corev1.SchemeGroupVersion)
+	}
+
+	encodeRespForAppsGV := func(obj runtime.Object) io.ReadCloser {
+		return encodeResp(obj, appsv1.SchemeGroupVersion)
+	}
+
+	tests := map[string]struct {
+		expNsListReq []string
+
+		dumpAllNs        bool
+		populateCmdFlags func(f *flag.FlagSet)
+	}{
+		"should dump default namespaces": {
+			expNsListReq: []string{"kube-system", "default"},
+
+			populateCmdFlags: func(f *flag.FlagSet) {
+				// use default options
+			},
+		},
+
+		"should dump requested namespaces": {
+			expNsListReq: []string{"qa", "production"},
+
+			populateCmdFlags: func(f *flag.FlagSet) {
+				f.Set("namespaces", "qa,production")
+			},
+		},
+
+		"should dump all available namespaces": {
+			expNsListReq: []string{"qa", "production", "test", "kube-system"},
+
+			dumpAllNs: true,
+			populateCmdFlags: func(f *flag.FlagSet) {
+				f.Set("all-namespaces", "true")
+			},
+		},
+	}
+
+	for tn, tc := range tests {
+		t.Run(tn, func(t *testing.T) {
+			expListReq := map[string]io.ReadCloser{
+				"/api/v1/nodes": encodeRespForCoreGV(&corev1.NodeList{}),
+			}
+
+			for _, ns := range tc.expNsListReq {
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/events", ns)] = encodeRespForCoreGV(&corev1.EventList{})
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/replicationcontrollers", ns)] = encodeRespForCoreGV(&corev1.ReplicationControllerList{})
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/services", ns)] = encodeRespForCoreGV(&corev1.ServiceList{})
+				expListReq[fmt.Sprintf("/api/v1/namespaces/%s/pods", ns)] = encodeRespForCoreGV(&corev1.PodList{})
+
+				expListReq[fmt.Sprintf("/apis/apps/v1/namespaces/%s/daemonsets", ns)] = encodeRespForAppsGV(&appsv1.DaemonSetList{})
+				expListReq[fmt.Sprintf("/apis/apps/v1/namespaces/%s/deployments", ns)] = encodeRespForAppsGV(&appsv1.DeploymentList{})
+				expListReq[fmt.Sprintf("/apis/apps/v1/namespaces/%s/replicasets", ns)] = encodeRespForAppsGV(&appsv1.ReplicaSetList{})
+			}
+
+			if tc.dumpAllNs { // register expected namespaces
+				var items []corev1.Namespace
+				for _, nsName := range tc.expNsListReq {
+					items = append(items, corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}})
+				}
+				expListReq["/api/v1/namespaces"] = encodeRespForCoreGV(&corev1.NamespaceList{
+					Items: items,
+				})
+			}
+
+			tf.Client = &fake.RESTClient{
+				NegotiatedSerializer: ns,
+				Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+					respBody, found := expListReq[req.URL.Path]
```

Comment:
There's only one time you care for the response body which is when requesting all namespaces, make a separate test for that particular case and return the list as desired. In case where you test against specific namespaces the response is sufficient to be `200 OK`. Then just focus on matching the URL. Make the test simpler to read, compare https://github.com/kubernetes/kubernetes/blob/f5f42df42bafb122be87c2d19d7c95d2f09998be/pkg/kubectl/cmd/expose/expose_test.go#L607-L619

Reply:
And please use readable variable names, `expListReq` is cryptic and that's wrong.